app/workers/sphinx_account_indexation_worker.rb

```diff
@@ -8,18 +8,14 @@
 # - User gets created, updated, deleted
 # - Cinstance gets created, updated, deleted
 class SphinxAccountIndexationWorker < SphinxIndexationWorker
-  def perform(model, id)
-    pk = model.primary_key
-    account = model.find_by(pk => id)
+  def perform(_model, id)
+    account = Account.searchable.find_by(id: id)
 
-    indices_for_model(model).each do |index|
-      if account && index.scope.find_by(pk => id)
-        reindex(index, account)
-      elsif account && !account.master?
-        delete_from_index(index, id, *account.buyers.pluck(:id))
-      else
-        delete_from_index(index, id)
-      end
+    if account
```

Comment:
Why can't you access the index? It seems I have access. When I place a breakpoint in this line:

```
    14:   binding.pry
 => 15:   if account
    16:     reindex(account)
    17:   else
```

```rb
[2] pry(#<SphinxAccountIndexationWorker>)> idxs = ThinkingSphinx::Configuration.instance.index_set_class.new(classes: [_model])
=> #<ThinkingSphinx::IndexSet:0x000000000e141e88
 @configuration=
  #<ThinkingSphinx::Configuration:0x00007fc0cc722b78
...
```

```rb
[4] pry(#<SphinxAccountIndexationWorker>)> idxs.count
=> 1
[5] pry(#<SphinxAccountIndexationWorker>)> idxs.first
=> #<ThinkingSphinx::RealTime::Index:0x00007fc0cc7353b8
 @attributes=
  [#<ThinkingSphinx::RealTime::Attribute:0x00007fc0cc734a30
    @column=
     #<ThinkingSphinx::ActiveRecord::Column:0x00007fc0cc734ad0
...
```

```rb
[6] pry(#<SphinxAccountIndexationWorker>)> idxs.first.scope
=> [#<Account:0x00007fc0cc6ff498
  id: 2,
  org_name: "Provider Name",
  org_legaladdress: "",
  created_at: Thu, 03 Aug 2023 15:28:30 UTC +00:00,
  updated_at: Thu, 03 Aug 2023 15:28:30 UTC +00:00,
...
```

```rb
[7] pry(#<SphinxAccountIndexationWorker>)> idxs.first.scope.size
=> 2
```

Is this not the same you see? Or is there another problem?

Reply:
From old version, what I see is that probably no index was found. Or no callbacks were available. I don't know why in some situations this happened because I couldn't locally or even in place reproduce with another image. So trying to be using supported APIs.